RssAgent: Migrate from FeedNormalizer to Feedjira

knu opened the pull request:
This replaces #957.

cantino commented on app/models/agents/rss_agent.rb:
```diff
@@ -139,21 +162,241 @@ def check_and_track(entry_id)
+    ENCLOSURE_ATTRS = %i[url type length]
+
+    unless dependencies_missing?
+      class AtomAuthor
```
Should we pull this into a separate file?

knu replied:
Maybe.  Would `app/models/agents/rss_agent/feedjira_extension.rb` be a good place?

cantino replied:
Or maybe in `lib/feedjira_extension.rb`?

cantino commented on app/models/agents/rss_agent.rb:
```diff
@@ -139,21 +162,241 @@ def check_and_track(entry_id)
+    def entry_data(entry)
+      description = entry.summary
+      content = entry.content || description
+      id = entry.try(:entry_id) || Digest::MD5.hexdigest(content || '')
```
Might be worth keeping `get_entry_id`, as #1553 modifies it.

cantino commented on spec/data_fixtures/onethingwell.atom:
```diff
@@ -208,8 +209,7 @@
             <title>Showgoers</title>
-            <description>&lt;a href="http://showgoers.tv/"&gt;Showgoers&lt;/a&gt;: &lt;blockquote&gt; &lt;p&gt;Showgoers is a Chrome browser extension to synchronize your Netflix player with someone else so that you can co-watch the same movie on different computers with no hassle. Syncing up your player is as easy as sharing a URL.&lt;/p&gt; &lt;/blockquote&gt;
-            </description>
+            <description>&lt;a href="http://showgoers.tv/" onmouseover="javascript:void(0)"&gt;Showgoers&lt;/a&gt;: &lt;blockquote&gt; &lt;p&gt;Showgoers is a Chrome browser extension to synchronize your Netflix player with someone else so that you can co-watch the same movie on different computers with no hassle. Syncing up your player is as easy as sharing a URL.&lt;/p&gt; &lt;/blockquote&gt;&l ...
```
So it no longer removes onmouseover / script, but it still safely escapes?

cantino replied:
Oh, this is with clean = false.

cantino commented:
Should we ask some of the people from #889 and #955 to test this?

cantino commented on app/models/agents/rss_agent.rb:
```diff
@@ -139,21 +160,80 @@ def check_and_track(entry_id)
     end
 
+    unless dependencies_missing?
+      require ''
```
`require ''`?

knu commented:
Oh, I didn't mean to push this yet.  I'll fix it later.

cantino commented:
Gotcha.

knu commented:
OK, I think this is finally getting ready for merge.  Since I've refactored and rewritten a large portion of what I pushed the last time, I've squashed everything into a single commit so the whole change gets reviewed again from the ground up.  Thank you for your review and testing!

cantino commented on app/models/agents/rss_agent.rb:
```diff
@@ -53,18 +52,43 @@ def default_options
       Events look like:
 
           {
+            "feed": {
```
Maybe

```
{
  "feed": {  # if you set `include_feed_info` to `true` 
```

knu replied:
Actually there was a note at the bottom but it was probably unnoticeable.  I've added a couple of additional notes about some fields, so it has hopefully gained its presence.

cantino commented on lib/feedjira_extension.rb:
```diff
@@ -0,0 +1,254 @@
+        addr = Mail::Address.new(content)
+      rescue
+        self.name = content
+      else
```
Cool, I didn't know you could do this on a `begin` block.

cantino commented:
This is a significant piece of work, thank you!

Want to ask folks in #889 and #955 to test this?

knu commented:
@cantino Yes, I'd much appreciate if people who have had trouble with the RssAgent could test this out and give some feedback.  /cc @marycanady, @schulzpin, @Jngai

knu commented:
I've been running this with various feeds for two weeks without a problem.  I'm afraid I was too late to come up when people had already found out their way to work around their problems, but is there any feedback from you folks?

I'd like to merge this soon, to move away from obsolete libraries and add features based on the new code base.

knu commented:
@cantino Is it about time we should merge this and move on?

cantino commented:
Yes @knu, I agree. Go for it!

knu commented:
Done!

dsander commented:
Awesome work 👏

knu commented:
Thanks.  Google Reader is long gone and it's been one year since Yahoo Pipes shut down, but feeds shall not die!

mcanady commented:
Thanks for your work on this--I'm in "set it and forget it" mode for this project so I can't test now, but will let you know when I do!
